src/overlays/actors/ovl_En_Suttari/z_en_suttari.c

```diff
@@ -21,110 +58,1507 @@ const ActorInit En_Suttari_InitVars = {
+            break;
+        case 7:
+            this->flags1 |= 0x20;
+            this->flags1 |= 0x80;
+            func_80BAC220(this, globalCtx);
+            break;
+        case 14:
+            this->flags1 |= 0x20;
+            this->flags1 |= 0x80;
+            func_80BABFD4(this, globalCtx);
+            break;
+        case 1:
+            Actor_MarkForDeath(&this->actor);
+            break;
+        case 2:
+            if (!(gSaveContext.weekEventReg[0x51] & 4)) {
+                this->flags1 |= 0x80;
+                this->actor.world.pos.x = -16.0f;
+                this->actor.world.pos.z = -16.0f;
+                this->actor.world.pos.y = 12.0f;
+                this->enFsn->flags |= ENFSN_HAGGLE;
+                this->actionFunc = func_80BADA08;
+            }
+            break;
+        case 4:
+            if (!(gSaveContext.weekEventReg[0x21] & 8)) {
+                if (this->animationIdx == 2 || this->animationIdx == 1) {
+                    this->animationIdx = 5;
+                    func_800BDC5C(&this->skelAnime, sAnimations, this->animationIdx);
+                }
+                this->flags1 |= 0x10;
+                if (this->flags2 & 2) {
+                    this->flags2 &= ~2;
+                }
+                if (!(player->stateFlags1 & 0x10000000)) {
+                    this->flags2 |= 0x10;
+                    player->stateFlags1 |= 0x10000000;
+                }
+                this->textId = 0x2A30;
+                func_801518B0(globalCtx, this->textId, &this->actor);
+                this->actionFunc = func_80BAD2B4;
+            }
+            break;
+    }
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.world.rot.y, 4, 0x1554);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Suttari_0x80BAA6D0/func_80BAB1A0.asm")
+void func_80BAC6E8(EnSuttari* this, GlobalContext* globalCtx) {
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600C240, &D_0600071C, this->jointTable, this->morphTable, 16);
+    this->actor.draw = EnSuttari_Draw;
+    this->actor.flags |= 1;
+    if (globalCtx->sceneNum == SCENE_IKANA) {
+        this->flags1 |= 1;
+        if (gSaveContext.day == 1 || gSaveContext.day == 2) {
+            this->animationIdx = 2;
+            func_800BDC5C(&this->skelAnime, sAnimations, this->animationIdx);
+            this->flags1 |= 0x80;
+            this->actionFunc = func_80BACA14;
+            return;
+        } else if ((gSaveContext.day == 3) && (gSaveContext.time <= CLOCK_TIME(19, 00)) &&
+                   !(gSaveContext.weekEventReg[0x3D] & 8) && !(gSaveContext.weekEventReg[0x21] & 8) &&
+                   (gSaveContext.weekEventReg[0x33] & 8)) {
+            this->animationIdx = 2;
+            func_800BDC5C(&this->skelAnime, sAnimations, this->animationIdx);
+            this->actionFunc = func_80BACEE0;
+            return;
+        }
+    } else if (globalCtx->sceneNum == SCENE_BACKTOWN) {
```

Comment:
This should work too, right?
```suggestion
    if (globalCtx->sceneNum == SCENE_IKANA) {
        this->flags1 |= 1;
        if (gSaveContext.day == 1 || gSaveContext.day == 2) {
            this->animationIdx = 2;
            func_800BDC5C(&this->skelAnime, sAnimations, this->animationIdx);
            this->flags1 |= 0x80;
            this->actionFunc = func_80BACA14;
        } else if ((gSaveContext.day == 3) && (gSaveContext.time <= CLOCK_TIME(19, 00)) &&
                   !(gSaveContext.weekEventReg[0x3D] & 8) && !(gSaveContext.weekEventReg[0x21] & 8) &&
                   (gSaveContext.weekEventReg[0x33] & 8)) {
            this->animationIdx = 2;
            func_800BDC5C(&this->skelAnime, sAnimations, this->animationIdx);
            this->actionFunc = func_80BACEE0;
        }
        return;
    } else if (globalCtx->sceneNum == SCENE_BACKTOWN) {
```

Reply:
No, the else case of this branch is actually what falls through to the `Actor_MarkForDeath` at the very end. From what I can tell, it is the only thing that falls all the way through. It just seems silly since an else block would accomplish the same thing and in my opinion be easier to read.